tests/test_project.py

```diff
@@ -1647,6 +1647,25 @@ def test_init_local_with_empty_path(repos_tmpdir):
     assert (repos_tmpdir / 'workspace' / 'subdir' / 'Kconfiglib').check(dir=1)
 
 
+def test_init_local_with_clone_option_failure(repos_tmpdir):
+    # Test that 'west init -l -o' errors out
+
+    west_tmpdir = repos_tmpdir / 'workspace'
+
+    with pytest.raises(subprocess.CalledProcessError):
+        cmd(['init', '-l', '-o=--depth=1', west_tmpdir])
+
+
+def test_init_with_clone_option_depth_one(repos_tmpdir):
+    # Test that 'west init -o=--depth=1' only clones depth 1
+
+    west_tmpdir = repos_tmpdir / 'workspace'
+
+    cmd(['init', '-o=--depth=1', west_tmpdir])
+    assert 1 == int(subprocess.check_output([GIT, 'rev-list', '--count', '--all'],
```

Comment:
`--all` does not make sense to me, it assumes `--single-branch` is the default so please replace it with `HEAD`. The fewer assumptions the better.

`--max-count=5` is a small and cheap optimization.

```suggestion
    assert 1 == int(subprocess.check_output([GIT, 'rev-list', '--count', '--max-count=5', 'HEAD'],
```

Reply:
```sh
tox  -- -k test_init_with_clone_option_depth_one
.pkg: _optional_hooks> python /home/usman/workdir/sls/zephyrproject/.venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python /home/usman/workdir/sls/zephyrproject/.venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: build_sdist> python /home/usman/workdir/sls/zephyrproject/.venv/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
py3: install_package> python -I -m pip install --force-reinstall --no-deps /home/usman/workdir/sls/zephyrproject/west/.tox/.tmp/package/48/west-1.2.99.tar.gz
py3: commands[0]> python -m pytest --cov-report=html --cov=west -k test_init_with_clone_option_depth_one '--basetemp=/home/usman/workdir/sls/zephyrproject/west/.tox/py3/tmp/pytest with space/'
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.11.2, pytest-8.3.3, pluggy-1.5.0
cachedir: .tox/py3/.pytest_cache
rootdir: /home/usman/workdir/sls/zephyrproject/west
configfile: pyproject.toml
plugins: cov-5.0.0
collected 200 items / 199 deselected / 1 selected                                                                                                                                                                 

tests/test_project.py .                                                                                                                                                                                     [100%]

---------- coverage: platform linux, python 3.11.2-final-0 -----------
Coverage HTML written to dir htmlcov


================================================================================== 1 passed, 199 deselected in 79.76s (0:01:19) ===================================================================================
py3: commands[1]> python -m flake8 --config=/home/usman/workdir/sls/zephyrproject/west/tox.ini /home/usman/workdir/sls/zephyrproject/west
py3: commands[2]> python -m mypy --config-file=/home/usman/workdir/sls/zephyrproject/west/tox.ini --package=west
src/west/app/project.py:1005: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Success: no issues found in 12 source files
  py3: OK (84.19=setup[3.07]+cmd[80.00,0.51,0.62] seconds)
  congratulations :) (84.23 seconds)
```